src/app/+data-and-analytics/academic-age-group/academic-age-group.component.ts

```diff
@@ -0,0 +1,151 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, QueryList, ViewChildren } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store, select } from '@ngrx/store';
+import { BehaviorSubject, Observable, Subject, filter, map, tap } from 'rxjs';
+
+import { SolrDocument } from '../../core/model/discovery';
+import { Filterable } from '../../core/model/request';
+import { DataAndAnalyticsView, DisplayView, OpKey } from '../../core/model/view';
+import { AppState } from '../../core/store';
+import { selectResourcesResearchAge } from '../../core/store/sdr';
+import { AcademicAge } from '../../core/store/sdr/sdr.reducer';
+import { fadeIn } from '../../shared/utilities/animation.utility';
+import { BarplotComponent, BarplotInput } from '../barplot/barplot.component';
+
+import * as fromSdr from '../../core/store/sdr/sdr.actions';
+
+const academicAgeGroupToBarplotInput = (academicAge: AcademicAge): BarplotInput => {
+  return {
+    label: academicAge.label,
+    data: academicAge.groups
+  } as BarplotInput;
+}
+
+const rk = 'Researchers';
+const pk = 'Publications';
+const apk = 'Average publications';
+
+@Component({
+  selector: 'scholars-academic-age-group',
+  templateUrl: './academic-age-group.component.html',
+  styleUrls: ['./academic-age-group.component.scss'],
+  animations: [fadeIn],
+})
+export class AcademicAgeGroupComponent implements OnInit, OnChanges {
+
+  @Input()
+  public organization: SolrDocument;
+
+  @Input()
+  public displayView: DisplayView;
+
+  @Input()
+  public dataAndAnalyticsView: DataAndAnalyticsView;
+
+  @Output()
+  public labelEvent: EventEmitter<string>;
+
+  @Input()
+  public upperLimitInYears = 40;
+
+  @Input()
+  public groupingIntervalInYears = 5;
+
+  @ViewChildren(BarplotComponent) barplots: QueryList<BarplotComponent>;
+
+  public maxOverride: Subject<number>;
+
+  public mean: Subject<number>;
+
+  public median: Subject<number>;
+
+  public academicAge: Observable<BarplotInput>;
+
+  public averagePubRateResearchAge: Observable<BarplotInput>;
+
+  constructor(private store: Store<AppState>, private route: ActivatedRoute) {
+    this.labelEvent = new EventEmitter<string>();
+    this.maxOverride = new BehaviorSubject<number>(undefined);
```

Comment:
I supposed I need to review typescript some more but, is `undefined` necessary here?

It seems like to me that if the parameter is missing (not passed) then it is `undefined` by default.

Example:
```ts
    this.maxOverride = new BehaviorSubject<number>();
```

I did some short research and I found this:
- https://dev.to/kildareflare/initialise-behavorsubject-correctly-hom

Reply:
Yes, it is required for use of `BehaviorSubject` for the initial value. In this case, I chose `undefined` to filter during observation. I will try to use `Subject`, which does not require a default, but in some cases the `BehaviorSubject` is necessary.